com.reprezen.swagedit/src/com/reprezen/swagedit/editor/SwaggerEditor.java

```diff
@@ -282,16 +282,16 @@ public void updateFoldingStructure(List<Position> positions) {
 	@Override
 	public void doSave(IProgressMonitor monitor) {
 		super.doSave(monitor);
-		validate();
+		validate(false);
 	}
 
 	@Override
 	public void doSaveAs() {
 		super.doSaveAs();
-		validate();
+		validate(false);
 	}
 
-	protected void validate() {
+	protected void validate(boolean onOpen) {
```

Comment:
`validate(true)` is used only once - on `doSetInput()`, in all other cases it's `validate(false)`. You could have left the default implementation like this:

```java
protected void validate() {
  validate(false);
}
protected void validate(boolean onOpen) { <...>}
```